src/lib/trips.js

```diff
@@ -39,7 +40,12 @@ export const transformSingleTimeSeriesToTrips = singleTimeseries => {
 
 export const transformTimeSeriesToTrips = geojsonTimeseries => {
   const allSeries = flatten(geojsonTimeseries.map(g => g.series))
-  return allSeries.map(transformSingleTimeSeriesToTrips)
+  const allSeriesWithGeojsonId = allSeries.map((s, i) => ({
+    ...s,
+    geojsonId: geojsonTimeseries[i]._id
+  }))
+
+  return allSeriesWithGeojsonId.map(transformSingleTimeSeriesToTrips)
```

Comment:
We map fourth times on almost the same object. Can't we do better? Like adding the `geojsonId` during the first map?

Reply:
Je n'ai pas réussi du premier coup non. Je note un todo de refactor, si j'ai le temps de repasser dessus avant la deadline je m'en occupe, sinon je le passerai au prochaine sprint